sov-modules/sov-state/src/value.rs

```diff
@@ -1,21 +1,15 @@
+use borsh::{BorshDeserialize, BorshSerialize};
 use std::marker::PhantomData;
 
 use crate::{Prefix, Storage, WorkingSet};
-use sovereign_sdk::serial::{Decode, Encode};
 use thiserror::Error;
 
 // SingletonKey is very similar to the unit type `()` i.e. it has only one value.
 // We provide a custom efficient Encode implementation for SingletonKey while Encode for `()`
 // is likely already implemented by an external library (like borsh), which is outside of our control.
-#[derive(Debug)]
+#[derive(Debug, BorshSerialize)]
 pub struct SingletonKey;
 
-impl Encode for SingletonKey {
```

Comment:
Maybe let's leave the code, see the comment and this issue:
https://github.com/near/borsh-rs/issues/19